WordPress/Classes/ViewRelated/Blog/SiteManagement/SiteTagsViewController.swift

```diff
@@ -246,11 +251,78 @@ extension SiteTagsViewController {
     }
 }
 
-// MARK: - Navigation
+// MARK: - Navigation to Tag details
 extension SiteTagsViewController {
-    fileprivate func navigate(to tag: PostTag) {
-        let singleTag = SiteTagViewController(blog: blog, tag: tag)
-        navigationController?.pushViewController(singleTag, animated: true)
+    fileprivate func navigate(to tag: PostTag?) {
+        let titleSectionHeader = NSLocalizedString("Tag", comment: "Section header for tag name in Tag Details View.")
+        let subtitleSectionHeader = NSLocalizedString("Description", comment: "Section header for tag name in Tag Details View.")
+        let content = SettingsTitleSubtitleController.Content(title: tag?.name,
+                                                              subtitle: tag?.tagDescription,
+                                                              titleHeader: titleSectionHeader,
+                                                              subtitleHeader: subtitleSectionHeader)
+        let confirmationContent = confirmation()
+        let tagDetailsView = SettingsTitleSubtitleController(content: content, confirmation: confirmationContent)
+
+        tagDetailsView.setAction { [weak self] updatedData in
+            self?.navigationController?.popViewController(animated: true)
+
+            guard let tag = tag else {
+                return
+            }
+
+            self?.delete(tag)
+        }
+
+        tagDetailsView.setUpdate { [weak self] updatedData in
+            guard let tag = tag else {
+                self?.addTag(data: updatedData)
+                return
+            }
+
+            guard let wasUpdated = self?.tagWasUpdated(tag: tag, updatedTag: updatedData), wasUpdated == true else {
```

Comment:
This might be a bit nitpicky of me, but as a small optimization we could skip the assignment and just check the return value is true.
`guard self?.tagWasUpdated(tag: tag, updatedTag: updatedData) == true else {`

But its really fine the way its written as well. :)